insights/parsers/installed_rpms.py

```diff
@@ -253,24 +253,21 @@ def __init__(self, *args, **kwargs):
 
     def parse_content(self, content):
         packages = defaultdict(list)
-        for line in get_active_lines(content, comment_char='COMMAND>'):
-            if line.startswith('error:') or line.startswith('warning:'):
-                self.errors.append(line)
+        all_active_lines = get_active_lines(content, comment_char='COMMAND>')
```

Comment:
This `get_active_lines` will **go through the whole file** to remove all the `comment_char` that appears in each line. For this parser, the input content is from the `rpm` command, there will never be multiple such `comment_char` comments.

The `"COMMAND>"` comment is for the `foreman-debug` archive only and it will always be the first line of the input content if it's a foreman archive.

So, better do not to use this `get_active_lines` function that iterates the who `installed_rpms` spec which is always a bit large. (save time aggressively :)

```python
if content and "COMMAND>" in content[0]:
    del content[0]  # or content = content[1:]
```

Reply:
Updated